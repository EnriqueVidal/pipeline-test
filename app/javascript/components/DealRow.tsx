import * as React from 'react';
import { Deal } from '../packs/useTabular';

interface Props {
  deal: Deal;
 }

const DealRow = ({ deal }: Props) => {
  const {
    deal_stage: dealStage,
    id,
    name,
    user,
    value,
  } = deal;

  const userName = [user.first_name, user.last_name].join(' ');
  const percentText = [dealStage.percent, '%'].join('');

  return (
    <tr>
      <td><strong>{id}</strong></td>
      <td>{userName}</td>
      <td>{name}</td>
      <td className="has-text-right">
        $
        {parseFloat(value).toFixed(2)}
      </td>
      <td>
        <span className="tag is-primary is-info">
          {dealStage.name}
        </span>
      </td>
      <td>
        <progress
          className="progress is-info"
          max={100}
          title={percentText}
          value={dealStage.percent}
        >
          {percentText}
        </progress>
      </td>
    </tr>
  );
};

export default DealRow;
