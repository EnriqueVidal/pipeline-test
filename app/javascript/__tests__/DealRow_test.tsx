import * as React from 'react';
import * as renderer from 'react-test-renderer';
import DealRow from '../components/DealRow';

describe('DealRow', () => {
  const persona5 = {
    id: 3,
    name: 'Persona 5',
    deal_stage: { id: 3, percent: 100, name: 'Wont!' },
    user: { id: 3, first_name: 'Ren', last_name: 'Amamiya' },
    value: '99999999999999',
  };

  it('matches snapshot', () => {
    const tree = renderer
      .create(<DealRow deal={persona5} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
