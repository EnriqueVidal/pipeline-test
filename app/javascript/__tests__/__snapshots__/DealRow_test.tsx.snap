// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DealRow matches snapshot 1`] = `
<tr>
  <td>
    <strong>
      3
    </strong>
  </td>
  <td>
    Ren Amamiya
  </td>
  <td>
    Persona 5
  </td>
  <td
    className="has-text-right"
  >
    $
    99999999999999.00
  </td>
  <td>
    <span
      className="tag is-primary is-info"
    >
      Wont!
    </span>
  </td>
  <td>
    <progress
      className="progress is-info"
      max={100}
      title="100%"
      value={100}
    >
      100%
    </progress>
  </td>
</tr>
`;
